locale/en/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,457 @@
+If you create fewer bytes for some users and more bytes for others, `crypto.randomBytes()` is another source of variation in Task lengths.
+
+### Task partitioning
+Tasks with variable time costs can harm the throughput of the Worker Pool.
+To minimize variation in Task times, as far as possible you should *partition* each Task into comparable-cost sub-Tasks.
+When each sub-Task completes it should submit the next sub-Task, and when the final sub-Task completes it should notify the submitter.
+
+To continue the `fs.readFile()` example, you should instead use `fs.read()` (manual partitioning) or `ReadStream` (automatically partitioned).
+
+The same principle applies to CPU-bound tasks; the `asyncAvg` example might be inappropriate for the Event Loop, but it is well suited to the Worker Pool.
+
+When you partition a Task into sub-Tasks, shorter Tasks expand into a small number of sub-Tasks, and longer Tasks expand into a larger number of sub-Tasks.
+Between each sub-Task of a longer Task, the Worker to which it was assigned can work on a sub-Task from another, shorter, Task, thus improving the overall Task throughput of the Worker Pool.
+
+Note that the number of sub-Tasks completed is not a useful metric for the throughput of the Worker Pool.
+Instead, concern yourself with the number of *Tasks* completed.
+
+### Avoiding Task partitioning
+Recall that the purpose of Task partitioning is to minimize the variation in Task times.
+If you can distinguish between shorter Tasks and longer Tasks (e.g. summing an array vs. sorting an array), you could create one Worker Pool for each class of Task.
+Routing shorter Tasks and longer Tasks to separate Worker Pools is another way to minimize Task time variation.
+
+In favor of this approach, partitioning Tasks incurs overhead (the costs of creating a Worker Pool Task representation and of manipulating the Worker Pool queue), and avoiding partitioning saves you the costs of additional trips to the Worker Pool.
+It also keeps you from making mistakes in partitioning your Tasks.
+
+The downside of this approach is that Workers in all of these Worker Pools will incur space and time overheads and will compete with each other for CPU time.
+Remember that each CPU-bound Task makes progress only while it is scheduled.
+As a result, you should only consider this approach after careful analysis.
+
+### Conclusions
+Whether you use only the Node Worker Pool or maintain separate Worker Pool(s), you should optimize the Task throughput of your Pool(s).
+
+To do this, minimize the variation in Task times by using Task partitioning.
+
+##  The risks of npm modules
+Node developers benefit tremendously from the [npm ecosystem](https://www.npmjs.com/), with hundreds of thousands of modules offering functionality to accelerate your development process.
+
+However, the vast majority of these modules are written by third-party developers and undergo no formal verification process.
```

Comment:
Does node core undergo a formal verification process? Mentioning it here makes it seem like it's in contrast to something else, but I'm not sure what that something is.

Reply:
You're right, "formal" isn't really the right word. It's more of a "many eyes on Node, few eyes on most 3rd-party modules" type situation. I'll change the wording.

Reply:
Addressed in ccd4f75.